workspaces/adventure-pack/goodies/kotlin/src/traverse_level_order/Test.kt

```diff
@@ -0,0 +1,32 @@
+package traverse_level_order
+
+import kotlin.test.*
+
+internal class LevelOrderTraversalTest {
+
+  @Test
+  fun itTraversesLevelOrderEmptyTree() {
+    val root: TreeNode? = null
+    assertContentEquals(
+        root.traverseLevelOrder().flatMap { it }.map { it.`val` }.toList(), emptyList<Int>())
+  }
+
+  @Test
+  fun itTraversesLevelOrderSingleNode() {
+    val root: TreeNode = TreeNode(0)
+    assertContentEquals(
+        root.traverseLevelOrder().flatMap { it }.map { it.`val` }.toList(), listOf<Int>(0))
```

Comment:
I would imagine `<Int>` for the list type is inferred from the use of 0 as the value?